ktor-client/ktor-client-core/common/src/io/ktor/client/features/NeedRetry.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.features
+
+import io.ktor.client.HttpClient
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.takeFrom
+import io.ktor.client.response.HttpReceivePipeline
+import io.ktor.client.response.HttpResponse
+import io.ktor.util.AttributeKey
+
+typealias NeedRetryHandler = suspend (response: HttpResponse) -> Boolean
```

Comment:
Would be nice to have `HttpRequestBuilder` for the future request here.

Reply:
great! I will change... Also, a friend of mine suggeste calling this type `RetryCondition` what do you think?

Reply:
`RetryCondition` looks perfect